packages/valory/skills/transaction_settlement_abci/payloads.py

```diff
@@ -41,226 +42,66 @@ def __str__(self) -> str:
-
-    @property
-    def randomness(self) -> str:
-        """Get the randomness."""
-        return self._randomness  # pragma: nocover
-
-    @property
-    def data(self) -> Dict:
-        """Get the data."""
-        return dict(round_id=self.round_id, randomness=self.randomness)
-
 
+@dataclass(frozen=True)
 class SelectKeeperPayload(BaseTxPayload):
     """Represent a transaction payload of type 'select_keeper'."""
 
+    keepers: str
     transaction_type = TransactionType.SELECT_KEEPER
 
-    def __init__(self, sender: str, keepers: str, **kwargs: Any) -> None:
-        """Initialize a 'select_keeper' transaction payload.
-
-        :param sender: the sender (Ethereum) address
-        :param keepers: the updated keepers
-        :param kwargs: the keyword arguments
-        """
-        super().__init__(sender, **kwargs)
-        self._keepers = keepers
-
-    @property
-    def keepers(self) -> str:
-        """Get the keeper."""
-        return self._keepers
-
-    @property
-    def data(self) -> Dict[str, str]:
-        """Get the data."""
-        return dict(keepers=self.keepers)
-
 
+@dataclass(frozen=True)
 class ValidatePayload(BaseTxPayload):
     """Represent a transaction payload of type 'validate'."""
 
+    vote: Optional[bool] = None
     transaction_type = TransactionType.VALIDATE
 
-    def __init__(self, sender: str, vote: Optional[bool] = None, **kwargs: Any) -> None:
-        """Initialize an 'validate' transaction payload.
-
-        :param sender: the sender (Ethereum) address
-        :param vote: the vote
-        :param kwargs: the keyword arguments
-        """
-        super().__init__(sender, **kwargs)
-        self._vote = vote
-
-    @property
-    def vote(self) -> Optional[bool]:
-        """Get the vote."""
-        return self._vote
-
-    @property
-    def data(self) -> Dict:
-        """Get the data."""
-        return dict(vote=self.vote) if self.vote is not None else {}
-
 
+@dataclass(frozen=True)
 class CheckTransactionHistoryPayload(BaseTxPayload):
     """Represent a transaction payload of type 'check'."""
 
+    verified_res: str
     transaction_type = TransactionType.CHECK
 
-    def __init__(self, sender: str, verified_res: str, **kwargs: Any) -> None:
-        """Initialize an 'check' transaction payload.
-
-        :param sender: the sender (Ethereum) address
-        :param verified_res: the verification result
-        :param kwargs: the keyword arguments
-        """
-        super().__init__(sender, **kwargs)
-        self._verified_res = verified_res
-
-    @property
-    def verified_res(self) -> str:
-        """Get the verified result."""
-        return self._verified_res
-
-    @property
-    def data(self) -> Dict:
-        """Get the data."""
-        return dict(verified_res=self.verified_res)
-
 
+@dataclass(frozen=True)
 class SynchronizeLateMessagesPayload(BaseTxPayload):
     """Represent a transaction payload of type 'synchronize'."""
 
+    tx_hashes: tuple[str]
```

Comment:
this may not be ideal as the order is irrelevant, however we need it to be hashable (for counting payloads the way we currently do it, may address later)

Reply:
`tx_hashes` is a string atm. This refactor is already complex enough, don't also change types of args please

Reply:
I agree with David. This PR could have been just the changes of the payload design, and then on another PR, you could update the tx settlement to showcase how the code is simplified (and improved since we do not need to serialize things such as the tx hashes anymore).